modular_skyrat/modules/clock_cult/code/components/brass_spreader.dm

```diff
@@ -0,0 +1,100 @@
+/// A component that spreads brass to a tile in [range] every [cooldown] seconds, converting everything on it into brass as well.
+/datum/component/brass_spreader
+	/// The range of which to spread brass
+	var/range = 5
+	/// The cooldown between spread attempts
+	var/cooldown = 5 SECONDS
+
+	COOLDOWN_DECLARE(turf_conversion_cooldown)
+
+
+/datum/component/brass_spreader/Initialize(range, cooldown)
+	if (!ismovable(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	START_PROCESSING(SSobj, src)
+
+	src.range = range
+	src.cooldown = cooldown
+
+
+/datum/component/brass_spreader/Destroy(force, silent)
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+
+/datum/component/brass_spreader/process(seconds_per_tick)
+	if(!COOLDOWN_FINISHED(src, turf_conversion_cooldown))
+		return
+
+	var/list/valid_turfs = list()
+	var/list/cult_turfs = list()
+
+	for(var/nearby_turf in circle_view_turfs(parent, range))
+		if(istype(nearby_turf, /turf/open/floor/bronze))
+			cult_turfs += nearby_turf
+			continue
+
+		var/static/list/blacklisted_spread_turfs = typecacheof(list(
+			/turf/open/floor/bronze,
+			/turf/open/space,
+			/turf/open/lava,
+			/turf/open/chasm,
+			/turf/open/misc/asteroid,
+		))
```

Comment:
I'm asking this because I'm not sure myself, but would it be better to have this inside of the function or where it currently is? I'd think that in it's current spot, it would be redefined every time it goes through the for loop.

Reply:
Nope, static vars are only initialized once

Reply:
gotcha